src/pytesmo/validation_framework/metric_calculators_adapters.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2020, TU Wien, Department of Geodesy and Geoinformation
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#   * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the following disclaimer in the
+#      documentation and/or other materials provided with the distribution.
+#    * Neither the name of the TU Wien, Department of Geodesy and
+#      Geoinformation nor the names of its contributors may be used to endorse
+#      or promote products derived from this software without specific prior
+#      written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL VIENNA UNIVERSITY OF TECHNOLOGY,
+# DEPARTMENT OF GEODESY AND GEOINFORMATION BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+# THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+"""
+Metric Calculator Adapters change how metrics are calculated by calling
+the `calc_metric` function of the adapted calculator instead of the unadapted
+version.
+"""
+
+from pytesmo.validation_framework.metric_calculators import (
+    PairwiseIntercomparisonMetrics,
+    TripleCollocationMetrics
+)
+import warnings
+import numpy as np
+
+
+class MonthsMetricsAdapter(object):
+    """
+    Adapt MetricCalculators to calculate metrics for groups across months
+    """
+
+    _supported = (
+        PairwiseIntercomparisonMetrics,
+        TripleCollocationMetrics,
+    )
+
+    def __init__(self, calculator, sets=None):
+        """
+        Add functionality to a metric calculator to calculate validation
+        metrics for subsets of certain months in a time series (e.g. seasonal).
+
+        Parameters
+        ----------
+        calculator : MetadataMetrics or any child of it (not Rolling metrics)
+        sets : dict, optional (default: None)
+            A dictionary consisting of a set name (which is added to the metric
+            name as a suffix) and the list of months that belong to that set.
+            If None is passed, we use 4 (seasonal) sets named after the fist
+            letter of each month used.
```

Comment:
Can you add an example here? For me it was not directly clear that list of months means a list of month numbers